tests/ignite/metrics/test_root_mean_squared_error.py

```diff
@@ -17,24 +18,48 @@ def test_zero_div():
 
 
 def test_compute():
+
     rmse = RootMeanSquaredError()
 
-    y_pred = torch.Tensor([[2.0], [-2.0]])
-    y = torch.zeros(2)
-    rmse.update((y_pred, y))
-    assert isinstance(rmse.compute(), float)
-    assert rmse.compute() == 2.0
+    def _test(y_pred, y, batch_size):
+        rmse.reset()
+        rmse.update((y_pred, y))
+
+        np_y = y.numpy().ravel()
+        np_y_pred = y_pred.numpy().ravel()
+
+        if batch_size > 1:
+            n_iters = y.shape[0] // batch_size + 1
+            for i in range(n_iters):
+                idx = i * batch_size
+                rmse.update((y_pred[idx : idx + batch_size], y[idx : idx + batch_size]))
+
+        np_res = np.sqrt(np.power((np_y - np_y_pred), 2.0).sum() / np_y.shape[0])
+        res = rmse.compute()
+
+        assert isinstance(res, float)
+        assert pytest.approx(res) == np_res
+
+    def get_test_cases():
+
+        test_cases = [
+            (torch.randint(0, 10, size=(50,)), torch.randint(0, 10, size=(50,)), 1),
+            (torch.randint(-10, 10, size=(50, 1)), torch.randint(-10, 10, size=(50, 1)), 1),
+            # updated batches
+            (torch.randint(0, 10, size=(50,)), torch.randint(0, 10, size=(50,)), 16),
+            (torch.randint(-10, 10, size=(50, 1)), torch.randint(-10, 10, size=(50, 1)), 16),
```

Comment:
Actually, why input tensors are int/long and not float as in the original implementation ?
@KickItLikeShika please pay attention on those details while porting things !